nixos/modules/services/web-apps/mattermost.nix

```diff
@@ -184,6 +184,25 @@ in
           .tar.gz files.
         '';
       };
+      secretFile = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Path to a json file containing secret config values which should
+          not be written into the Nix store. If it is not null (the default)
+          and <option>services.mattermost.mutableConfig"</option>
+          is set to <literal>false</literal>, then the mattermost service will
+          join the options declared in this file into its config on startup. See
+          <link xlink:href="https://docs.mattermost.com/administration/config-settings.html"/>
+          for the formatting and available options.
+
+          Note that this merging will result in overwriting options which are
+          set declaratively.
+
+          if <option>services.mattermost.mutableConfig"</option>
+          is set to <literal>true</literal>, then this option has no effect.
+        '';
+      };
 
```

Comment:
```suggestion
      environmentFile = mkOption {
        type = types.nullOr types.path;
        default = null;
        description = ''
          Environment file (see <literal>systemd.exec(5)</literal>
          "EnvironmentFile=" section for the syntax) to define variables for
          Mattermost. Environment variables will override settings in config.json. See 
          https://docs.mattermost.com/configure/configuration-settings.html#environment-variables for reference
        '';
      };
```

Reply:
Unit further down would need, but I can't suggest the change there.
```nix
        EnvironmentFile = [  ] ++ optional (cfg.environmentFile != null) cfg.environmentFile;

```

Reply:
That's indeed be a simpler solution (though tbh I quite like having the secret file having essentially the same nice hierarchical structure as the config itself, but that's just personal preference).

I'm not sure why I didn't do it that way — probably I wasn't aware mattermost could be configured via environment variables, or perhaps that even got added in the year since I originally wrote this PR?